R/joss_article.R

```diff
@@ -0,0 +1,91 @@
+#' Journal of Open Source Software (JOSS) format.
+#'
+#' Format for creating a Journal of Open Source Software (JOSS) or Journal of
+#' Open Source Education (JOSE) articles. Adapted
+#' from \url{https://github.com/openjournals/whedon}. As these journals take
+#' articles as markdown, this format can be used to generate markdown from
+#' R Markdown and to locally preview how the article will appear as PDF.
+#'
+#' @inheritParams rmarkdown::pdf_document
+#' @param journal one of "JOSS" or"JOSE"
+#' @param keep_md Whether to retain the intermediate markdown and images.
+#'   Defaults to TRUE.
+#' @param doi,year,volume,issue,page,submitted,published,review_url,repository,archive_doi
+#'   Article metadata, all optional for generating PDF preview, auto-generated
+#'   on submission.
+#' @param latex_engine,... Arguments passed to \code{rmarkdown::pdf_document}
+#' @export
+joss_article <- function(journal = "JOSS",
+                         keep_md = TRUE,
+                         doi = "",
+                         year = "",
+                         volume = "",
+                         issue = "",
+                         page = "",
+                         submitted = "",
+                         published = "",
+                         review_url = "",
+                         repository = "",
+                         archive_doi = "",
+                         latex_engine = "xelatex",
+                         ...) {
+
+  rmarkdown::pandoc_available('2.2', TRUE)
+
+  logo_path <- find_resource("joss_article",
+                             paste0(journal, "-logo.png"))
+  journalname <- ifelse(journal == "JOSS",
+                        "Journal of Open Source Software",
+                        "Journal of Open Source Education")
+
+  template_variables <- c(
```

Comment:
How about setting these variables in the YAML frontmatter of the Rmd document? That way, you won't have to worry about escaping certain special characters in the values of these variables (although I guess it would be unlikely that these values could contain characters like double quotes). It will also make the R function much simpler.

Reply:
I wanted to avoid needing these variables in the YAML so that a standard `paper.md` would compile without setting them, but now I realized that the document will compile without most of them.  The logo and journal name still make sense to define here, but I can drop the rest.